### DISCUSSION TITLE
Help: Failed when opening source(s)...

### DISCUSSION THREAD

**Discussion by MikeMcCollister** [microsoft/winget-cli#3258] [Q&A] [answered]
I'm trying to install git on my computer as admin. What I do is type the following from an admin Windows Terminal:

`winget install --id Git.Git -e --source winget`

I get this error:

```
Failed when opening source(s); try the 'source reset' command if the problem persists.
An unexpected error occurred while executing the command:
0x8a15000f : Data required by the source is missing

```
Based on some searching that I have done I've tried the following commands:

```
winget source reset --force
winget source update
```
But they have not made a difference.

Any ideas what to do?

I'm running winget version v1.4.10173.

Thanks,

Mike

**Comment by Trenly** [comment]
If you download the file from https://cdn.winget.microsoft.com/cache/source.msix how many KB is it?

**Reply by MikeMcCollister** [reply]
On my admin account the size of the file is 7 KB. On my standard user account, the size is 6.84 MB.

**Reply by Trenly** [reply, marked as answer]
Hmm, that means it isn't a CDN issue. If you install that MSIX manually using `Add-AppxPackage -Path https://cdn.winget.microsoft.com/cache/source.msix` or by double clicking to install, that would force the winget source to be updated; That might narrow down whether it's the winget source or msstore source

**Reply by MikeMcCollister** [reply]
OK, I ran the command `Add-AppxPackage -Path https://cdn.winget.microsoft.com/cache/source.msix` and that fixed the issue.

Any ideas how my system got into that bad state?

**Reply by Trenly** [reply]
Appx and Msix packages are by default user-scoped and the data is typically to the scope of the user that installed them. I'm completely guessing here, but I believe that this may have caused the admin account to see that winget was installed and be able to run winget, but because the source msix package was last installed as a user, the admin instance of winget couldn’t open the underlying data

**Reply by zaadevofc** [reply]
yeahh thanks bruh this actualy work

**Comment by PhotoAtomic** [comment]
just change region in windows settings, probably your local CDN is having issue, use a CDN of a different region
in windows 11 -> settings -> time and language -> region -> country or region and choose a different one (US maybe)

**Reply by FledaJohansson** [reply]
This fixed it for me

**Comment by suren-c** [comment]
Just try executing the command in powershell as administrator. Should solve the problem.

**Comment by santagada** [comment]
is this answered? Feels like the system not being able to detect, inform or fix itself is a bug and having to meddle with random files on the system to get it unbroken is not really a fix but a quick hack no?

I guess what is needed is for winget to be better at reporting what is the actual error... being able to fix itself winget source reset --force should do it, and not get in this state.

**Comment by AmodeusR** [comment]
Just to help those who might come to have this issue, It's a winget deprecated version problem. You just need to update it, and the way I did was downloading the AppInstaller from the link below and then following the described steps.

https://aka.ms/getwinget

```bash
# Download *.msixbundle (from linke above)
# cd <download location>
Add-AppxPackage -Path '.\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'

winget -v

#v1.9.25200

```

**Comment by asmaier** [comment]
I got this error message, because I had no internet connection in my VM. So check your internet connection first.
